packages/proxy/lib/http/response-middleware.ts

```diff
@@ -283,10 +256,10 @@ const SetInjectionLevel: ResponseMiddleware = function () {
       return 'partial'
     }
 
-    const isSecondaryOrigin = this.remoteStates.isSecondaryOrigin(this.req.proxiedUrl)
+    const isCrossOrigin = !reqMatchesOriginPolicy(this.req, this.remoteStates.getPrimary())
     const isAUTFrame = this.req.isAUTFrame
 
-    if (this.config.experimentalSessionAndOrigin && isSecondaryOrigin && isAUTFrame && (isHTML || isRenderedHTML)) {
+    if (this.config.experimentalSessionAndOrigin && isCrossOrigin && isAUTFrame && (isHTML || isRenderedHTML)) {
```

Comment:
So this injects no matter what for cross origin visits when loading AUT HTML and we no longer hold up the server? I like it!

Reply:
Yep, if the request is cross origin and html, we're going to inject, but only create a spec bridge on demand.